pkg/model/mselection/heaviest.go

```diff
@@ -176,38 +268,16 @@ func (s *HeaviestSelector) OnNewSolidBundle(bndl *tangle.Bundle) {
 
 	// we have to check the below max depth criteria for approvees that do not reference our future cone.
 	// if all the unknown approvees do not fail the below max depth criteria, the tip is valid
-	if len(approveeHashes) > 0 {
-		lsmi := tangle.GetSolidMilestoneIndex()
-
-		for approveeHash := range approveeHashes {
-			var approveeORTSI milestone.Index
-
-			if tangle.SolidEntryPointsContain(hornet.Hash(approveeHash)) {
-				// if the approvee is an solid entry point, use the EntryPointIndex as ORTSI
-				approveeORTSI = tangle.GetSnapshotInfo().EntryPointIndex
-			} else {
-				cachedApproveeTx := tangle.GetCachedTransactionOrNil(hornet.Hash(approveeHash)) // tx +1
-				if cachedApproveeTx == nil {
-					panic(fmt.Sprintf("transaction not found: %v", hornet.Hash(approveeHash).Trytes()))
-				}
-
-				_, approveeORTSI = s.getTransactionRootSnapshotIndexes(cachedApproveeTx.Retain(), lsmi) // tx +1
-				cachedApproveeTx.Release(true)
-			}
-
-			// if the confirmationIdx to LSMI delta of the approvee is equal or greater belowMaxDepth, the tip is invalid.
-			// "equal" is important because the next milestone would reference this transaction.
-			if lsmi-approveeORTSI >= belowMaxDepth {
-				return
-			}
-		}
+	if !checkBelowMaxDepth(approveeHashes) {
```

Comment:
What is generally, the current philosophy with BMD? Should transactions that became solid, but are below max depth really trigger this event? Or could they be filtered out earlier? For what would BMD transactions be relevant are they not just invalid and be discarded?
Not having to check for below max depth would also make the tip selector code much cleaner as the BMD check is actually something not really related to this particular tip selection.